### PR TITLE
feat(theme-builder): core: fix alpha normalisation issues and add tests

### PR DESCRIPTION
**Pull request by amery**
## Summary by CodeRabbit

- **New Features**
  - Introduced utility functions for consistent alpha value normalization and color input handling across color conversion tools.
  - Added new functions for converting and splitting color formats, improving color processing capabilities.

- **Tests**
  - Expanded and reorganized the test suite for color conversions, alpha handling, and edge cases, significantly increasing test coverage and robustness.

- **Chores**
  - Added a dedicated Vitest configuration file with code coverage reporting for the theme builder package.

### CODE REVIEW

File: packages/theme-builder/src/core/colors.ts [hunk 1]
```diff
@@ -70,6 +70,46 @@ export interface HctColor {
...
+ * @param a - Optional alpha value to normalize
+ * @returns Normalized alpha value or undefined if input is undefined
+ */
+export function normalizeAlpha(a?: number): number | undefined {
+  if (a === undefined) {
+    return undefined;
+  }
+  return a > 1 ? a / 255 : a;
+}
+
```
[the remark this answers is not in the document]
Comment (amery) [reply packages/theme-builder/src/core/colors.ts:84]: good point

[the remark this answers is not in the document]
Comment (amery) [reply packages/theme-builder/src/core/colors.ts]: PR amended

**Comment by amery** [comment]
@coderabbitai full review
